flow360/component/simulation/entity_info.py

```diff
@@ -0,0 +1,142 @@
+"""Desearlizer for entity info retrieved from asset metadata pipeline."""
+
+from abc import ABCMeta, abstractmethod
+from typing import List, Literal, Union
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.entity_registry import EntityRegistry
+from flow360.component.simulation.primitives import Edge, GenericVolume, Surface
+
+
+class EntityInfoModel(pd.BaseModel, metaclass=ABCMeta):
+    """Base model for asset entity info JSON"""
+
+    model_config = pd.ConfigDict(
+        ##:: Pydantic kwargs
+        extra="ignore",
+        frozen=False,
+        populate_by_name=True,
+        validate_assignment=True,
+        validate_default=True,
+    )
+
+    @abstractmethod
+    def get_full_list_of_boundaries(self, attribute_name: str = None) -> list:
+        """
+        Helper function.
+        Get the full list of boundary names. If it is geometry then use supplied attribute name to get the list.
+        """
+
+
+class GeometryEntityInfo(EntityInfoModel):
+    """Data model for geometry entityInfo.json"""
+
+    type_name: Literal["GeometryEntityInfo"] = pd.Field("GeometryEntityInfo", frozen=True)
+    face_ids: list[str] = pd.Field(
+        [],
+        description="A full list of faceIDs/model IDs that appear in the geometry.",
+        alias="faceIDs",
+    )
+    face_attribute_names: List[str] = pd.Field(
+        [],
+        description="A full list of attribute names that the user can"
+        "select to achieve grouping of faces. It has same length as `grouped_faces`",
+        alias="faceAttributeNames",
+    )
+    grouped_faces: List[List[Surface]] = pd.Field(
+        [[]],
+        description="The resulting list "
+        "of `Surface` entities after grouping using the attribute name.",
+        alias="groupedFaces",
+    )
+
+    edge_ids: list[str] = pd.Field(
+        [],
+        description="A full list of edgeIDs/model IDs that appear in the geometry.",
+        alias="edgeIDs",
+    )
+    edge_attribute_names: List[str] = pd.Field(
+        [],
+        description="A full list of attribute names that the user can"
+        "select to achieve grouping of edges. It has same length as `grouped_edges`",
+        alias="edgeAttributeNames",
+    )
+    grouped_edges: List[List[Edge]] = pd.Field(
+        [[]],
+        description="The resulting list "
+        "of `Edge` entities after grouping using the attribute name.",
+        alias="groupedEdges",
+    )
+
+    def group_items_with_given_tag(
+        self,
+        entity_type_name: Literal["face", "edge"],
+        attribute_name: str,
+        registry: EntityRegistry,
+    ) -> List[Union[Surface, Edge]]:
+        """
+        Group items with given attribute_name.
+        """
+        entity_list = self._get_full_list_of_given_entity_type(attribute_name, entity_type_name)
+        for item in entity_list:
+            registry.register(item)
+        return registry
+
+    def _get_full_list_of_given_entity_type(
```

Comment:
I think name of this function is not good. `full_list` and `given` are redundant.
```suggestion
    def _get_list_of_entities(
```

Reply:
Changed